jquery.overlay.js

```diff
@@ -211,23 +216,40 @@
           // Style attribute's string
           style = 'background-color:' + strategy.css['background-color'];
 
-          text.contents().each(function () {
-            var text, html, str, prevIndex;
-            if (this.nodeType != Node.TEXT_NODE) return;
-            text = this.textContent;
+          if (this.allowOverlapping) {
+            var textContent, prevIndex, str, html;
+            textContent = text.html()
             html = '';
             for (prevIndex = match.lastIndex = 0;; prevIndex = match.lastIndex) {
-              str = match.exec(text);
+              str = match.exec(textContent);
               if (!str) {
-                if (prevIndex) html += escape(text.substr(prevIndex));
+                if (prevIndex) html += textContent.substr(prevIndex);
                 break;
               }
               str = str[0];
-              html += escape(text.substr(prevIndex, match.lastIndex - prevIndex - str.length));
-              html += '<span style="' + style + '">' + escape(str) + '</span>';
+              html += textContent.substr(prevIndex, match.lastIndex - prevIndex - str.length);
+              html += '<span style="' + style + '">' + str + '</span>';
             };
-            if (prevIndex) $(this).replaceWith(html);
-          });
+            if (prevIndex) text.html(html);
+          } else {
+            text.contents().each(function () {
+              var text, html, str, prevIndex;
+              if (this.nodeType != Node.TEXT_NODE) return;
+              text = this.textContent;
+              html = '';
+              for (prevIndex = match.lastIndex = 0; ; prevIndex = match.lastIndex) {
```

Comment:
well this one is the existing code, just indented, but i did also use the same thing above. i'm not inclined to dry it up though, just because i wouldn't gain a huge amount in neatness and it would be involved enough to make maintenance substantially harder (i think the escapes are really what tip that balance)